src/coredump/coredumpctl.c

```diff
@@ -920,6 +978,10 @@ static int save_core(sd_journal *j, FILE *file, char **path, bool *unlink_temp)
                         return r;
                 assert(r > 0);
 
+                free_and_replace(filename, (char *) { path_join(j->prefix, filename) });
+                if (!filename)
+                        return log_oom();
+
                 if (access(filename, R_OK) < 0)
```

Comment:
so strictly speaking access() like this isn't really ok, because it would resolve symlinks relative to host than to arg_root.

You'd have to use chase_symlinks_and_access() here.

Reply:
@poettering Do I still need to do that now that I use `chase_symlinks` (in `make_filename_absolute`) to get the absolute filename?

Reply:
yes. using chase_symlinks_and_access() makes this less racy, in case somebody modifies the path between the chase_symlinks() and the access(). the big benefit of chase_symlinks_and_access() is that it internally pins every inode it accesses via an fd, and thus definitely executes the access() on that inode instead of an inode an attacker might have swapped out between the time we resolved the path and issues the access().

For new code we really prefer to use these less racy constructs.